### PR TITLE
Changes to include optional ADP calculation with council level basefile

### PR DESCRIPTION
**Pull request by vaelliott**
If you want an indicator to test the changes then you could use alcohol_stays - the 11to25 years is CA level.
The population parameter would need to be updated to
              pop = "ADP_pop_11to25"

### CODE REVIEW

File: 1.indicator_analysis.R [hunk 1]
```diff
@@ -106,7 +114,7 @@ analyze_first <- function(filename, geography = c("council", "datazone11"),
       group_by(code, year, sex_grp, age_grp) %>% summarise_all(sum, na.rm =T) %>% ungroup()
 
     } else if (measure == "stdrate" & geography == "council") {
-      data_indicator <- data_indicator %>% gather(geolevel, code, ca, hb, scotland) %>% 
+      data_indicator <- data_indicator %>% gather(geolevel, code, ca, hb:scotland) %>%
```
Comment (vaelliott) [inline 1.indicator_analysis.R]: I'm hoping I can copy the trick you used in line 104/112 where the command will be flexible enough to cope with the presence or absence of a column for ADP

File: 1.indicator_analysis.R [hunk 2]
```diff
@@ -86,16 +86,24 @@ analyze_first <- function(filename, geography = c("council", "datazone11"),
...
       geo_lookup <- readRDS(paste0(lookups, 'Geography/DataZone11_All_Geographies_Lookup.rds')) %>% 
         select(ca2019, hb2019) %>% distinct %>%  rename(ca = ca2019, hb = hb2019)
       
       ## Matching with geography lookup.
       data_indicator <- left_join(x=data_indicator, y=geo_lookup, c("ca")) %>% 
         mutate(scotland = as.factor("S00000001")) # adding Scotland
       
+    } else if (geography == "council" & adp==TRUE) {
+      geo_lookup <- readRDS(paste0(lookups, 'Geography/DataZone11_All_Geographies_Lookup.rds')) %>% 
+        select(ca2019, hb2019, adp) %>% distinct %>%  rename(ca = ca2019, hb = hb2019)
```
Comment (jvillacampa) [inline 1.indicator_analysis.R]: There is a bit of repetition going on here that makes it difficult to see what is different between the different ifs statements. I would do a nested if statement to make it more readable, but this is my personal perference, so ignore if you want:

`
geo_lookup <- readRDS(paste0(lookups, "Geography/DataZone11_All_Geographies_Lookup.rds")) 

    # Merging data with lookup depending on geography base
    if(geography == "datazone11") {
      
      ## Matching with geography lookup.
      data_indicator <- left_join(x=data_indicator, y=geo_lookup, c("datazone" = "datazone2011")) %>% 
        mutate(scotland = as.factor("S00000001")) # adding Scotland
      
    } else if (geography == "council" ) {

      if (adp == FALSE) { #different variables required if ADP included
        geo_lookup <- geo_lookup %>% select(ca2019, hb2019) 
      } else if (adp==TRUE) {
        geo_lookup <- geo_lookup %>% select(ca2019, hb2019, adp)
      }
      
      geo_lookup <- geo_lookup %>% distinct %>%  rename(ca = ca2019, hb = hb2019)
      
      ## Matching with geography lookup.
      data_indicator <- left_join(x=data_indicator, y=geo_lookup, c("ca")) %>% 
        mutate(scotland = as.factor("S00000001")) # adding Scotland
    }
    `

File: 1.indicator_analysis.R (hunk 1 above)
Comment (jvillacampa) [reply 1.indicator_analysis.R]: that looks good

File: 1.indicator_analysis.R (hunk 2 above)
Comment (vaelliott) [reply 1.indicator_analysis.R]: your way is better....much clearer so i've updated code

**Pull request merged**